src/renderer/entities/transactionBuilder/lib/leaf.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  CallBuilder, CallBuilding,
+  TransactionBuilder,
+  TransactionVisitor
+} from "@entities/transactionBuilder/model/transaction-builder";
+import {AccountInWallet} from "@shared/core/types/wallet";
+import {ApiPromise} from "@polkadot/api";
+import {BaseTxInfo, methods, OptionsWithMeta, UnsignedTransaction} from "@substrate/txwrapper-polkadot";
+import {SubmittableExtrinsic} from "@polkadot/api/types";
+import {Exception} from "@zxing/library";
+
+export class LeafTransactionBuilder implements TransactionBuilder, CallBuilder {
+
+  effectiveCallBuilder: CallBuilder;
+  #calls: CallBuilding[]
+
+  api: ApiPromise
+
+  readonly accountInWallet: AccountInWallet
+
+  constructor(api: ApiPromise, accountInWallet: AccountInWallet) {
+    this.effectiveCallBuilder = this
+    this.#calls = []
+    this.api = api
+
+    this.accountInWallet = accountInWallet
+  }
+
+  visit(_: TransactionVisitor): void {
+    // nothing interesting to visit in the leaf
+  }
+
+  addCall(call: CallBuilding): void {
+    this.#calls.push(call)
+  }
+
+  resetCalls(): void {
+    this.#calls = []
+  }
+
+  setCall(call: CallBuilding): void {
```

Comment:
What's the difference between `addCall` & `setCall`?

Reply:
`setCall` is the shortcut for `reseset + addCall`. In general those methods are just my idea what might potentially be needed; it is quite easy to alter them